web/iframe/compatibility.js

```diff
@@ -55,9 +55,18 @@
         base = m[0];
       }
     }
-    this.href = base + url;
+    this._setURL(base + url);
   }
   newURL.prototype = {
+    _setURL: function (url) {
+      this.href = url;
+      var m = /^(\w+:)\/\/([^:/]+)(:([0-9]+))?/.exec(url.toLowerCase());
```

Comment:
It is code for the legacy browsers (as explained in the top of the file or function), I'll add a simple comment here.